src/Plugin/migrate/process/FixUrl.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\localgov_legacy_migration\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Extract and fix URLs from a text field.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "fix_url"
+ * )
+ *
+ * @code
+ * localgov_directory_website/uri:
+ *   plugin: fix_url
+ *   source: field_website_plain
+ * @endcode
+ */
+class FixUrl extends ProcessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    if (isset($value['value'])) {
+      $url = $value['value'];
+
+      // This is very crude, but will ensure $url passes Drupal URL validation.
+      if (substr($url, 0, 1) === '/') {
+        $url = \Drupal::request()->getSchemeAndHttpHost() . $url;
+      }
+      elseif (!preg_match('/^(http|https):\/\//i', $url)) {
+        $url = 'http://' . $url;
```

Comment:
Is that what this code is doing? : If the URL does not start with "http" or "https" then append **http://** to the URL.

Reply:
The source field is plain text and the destination is a Drupal link and this does the bare minimum to ensure the contents of the field pass the URL validation. We might want to beef this up and reject anything that's not a link, but I'm working on the assumption that as not all fields map one to one all directory content will need manually reviewing after the migration and so this sort of thing can be cleaned up later.

Reply:
Okay, thanks.